rust/connlib/tunnel/src/device_channel/tun/closeable.rs

```diff
@@ -0,0 +1,35 @@
+use std::os::fd::{AsRawFd, RawFd};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+#[derive(Debug)]
+pub(crate) struct Closeable<T> {
+    closed: AtomicBool,
+    value: T,
+}
+
+impl<T: AsRawFd> AsRawFd for Closeable<T> {
+    fn as_raw_fd(&self) -> RawFd {
+        self.value.as_raw_fd()
+    }
+}
+
+impl<T: Copy> Closeable<T> {
```

Comment:
Why enforce `Copy` here? I'd suggest you pass a reference to `f` instead and if the caller constructed `Closeable` with a type that is copy, they can always move out of it themselves. Or make the type not generic in the first place if we only ever use it with a single one (haven't checked).